### PR TITLE
REL-2032: Do not submit to partners with zero requested time

### PR DESCRIPTION
**Pull request by cquiroz**
Updated a problem robot to forbid submissions when any partner's request is zero

Th relevant commit is cc574d4, the rest is code improvements and formatting

### CODE REVIEW

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala [hunk 1]
```diff
@@ -459,10 +459,20 @@ object TimeProblems {
...
-}
 
-import ProblemRobot._
-import TimeProblems._
+  // REL-2032 Check that none of the requested times per partner are zero
+  def partnerZeroTimeRequest(p: Proposal, s:ShellAdvisor): List[ProblemRobot.Problem] = p.proposalClass match {
+    case g:GeminiNormalProposalClass =>
+      val probs = for {
+          sub <- g.subs.swap.right
+        } yield for {
```
Comment (sraaphorst) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: Can someone explain to me why you would do a:
`for { ... } yield for { ... } yield ...`?
I'm not sure what this does. Is it equivalent to a nested loop in this case so the result is an iterable of iterable of Problem?

**Comment by sraaphorst** [comment]
I'm not familiar with the logic, but the cleanup looks great. :+1:

File: bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala (hunk 1 above)
Comment (cquiroz) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: The first one gets you out of the `Either`, the second iterates over sub which is a `List[NgoSubmission]`

Comment (tpolecat) [inline bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/robot/ProblemRobot.scala]: Might be easier to read without the `for` syntax. It's just 

```scala
g.subs.swap.right.map { sub =>
  sub.filter(_.request.time.value <= 0.0).map(ps => new Problem(...)))
}
```

**Comment by swalker2m** [comment]
:+1:

**Pull request merged**